forte/data/entry_type_generator.py

```diff
@@ -10,6 +11,15 @@ def _get_type_attributes():
     return type_attributes
 
 
+def _get_entry_attribute_by_class(input_entry_class: str):
+    modname, _, clsname = input_entry_class.rpartition(".")
+    mod = importlib.import_module(modname)
+    try:
+        return list(getattr(mod, clsname).__annotations__.keys())
```

Comment:
So we are relying on the class annotations to find the attributes? This might be OK, but this means we need to enforce all entry classes are generated from our `JSON` files (previously, an alternative way is that one can actually modify the classes). 

That being said, this solution may still be OK. I just would like to confirm is this the only way to do this? If this is the case, we need to make a note to make sure our documentation enforce typing very strictly.

Reply:
Now we have two ways to find the attributes, one is __annotations__, the other one is __dataclass_fields__.  like the below image shows.
<img width="729" alt="image" src="https://user-images.githubusercontent.com/11881183/156533717-a606c199-fa44-4ece-8173-f212663a6d2e.png">

However, when we use __dataclass_fields__, It will also find the attributes of the parent class. as the below image shows.
<img width="833" alt="image" src="https://user-images.githubusercontent.com/11881183/156534117-dcca3f00-5e7a-40d4-a5eb-83afc18df7d1.png">

So from my side,  __annotations__ may be the only way to do this.

Reply:
hmm, interesting. So `dataclass` fields might actually be better (less relying to type annotation). And we might actually want everything from all the ancestor classes. @qinzzz do you need the attributes from all the ancestors.

Reply:
If our Entry name is 'ft.onto.base_ontology.Token' or similar, we can use 'annotation' or '__dataclass_fields__' to get all the attributes.  And the difference is shown above.

However, when I use the entry name as 'entry_type_generator_test.TokenTest' in the test. The class does not have the '__dataclass_fields__' parameter. So in this case, we can non use '__dataclass_fields__' to do this.  As the below images shows.

<img width="696" alt="image" src="https://user-images.githubusercontent.com/11881183/157024620-075d1875-4683-40e1-9a29-17b5c9c65d49.png">
![image](https://user-images.githubusercontent.com/11881183/157024695-8c3aa6df-fda7-425b-bc43-128457ec9d73.png)

So 'annotations' may be the only way to do this.